app/src/marketplace/containers/EditProductPage2/EditControllerProvider.jsx

```diff
@@ -149,16 +153,52 @@ function useEditController(product: Product) {
         product,
     ])
 
+    const deployContract = useCallback(async () => {
+        if (product && product.id) {
```

Comment:
This could just launch the modal and get the result of the deploy, like
`const result = await deployModal.open()`

then based on the result, show notification. I assume we redirect back to product page after.